### PR TITLE
Added prop to override internal setting of active state in TopLinkItem

### PR DESCRIPTION
**Pull request by mrazian85**
**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

While doing the work to re-factor the Left Navigation on the dpv3 from an Astro Component back into a React Component, I encountered an issue whereby the way in which the Component internally sets which link should be active was causing a delayed visual effect when the Component is re-rendered after navigating to a new page in Astro.

After conferring with @enrique-prado, we decided the best solution was to provide a prop to override the internal Left Navigation logic and allow for custom class names to be passed in instead, thus providing a way to bypass setting active state via `useEffect`, and allowing for a smoother visual effect on render. This PR is my implementation of this idea.

**EXAMPLE FUNCTIONALITY GIF:**
Left Nav without Override:

Left Nav without Override

Left Nav with Override:

Left Nav with Override

### CODE REVIEW

**Comment by joe-s-avaya** [comment]
you deleted all the badges? that's a no-no

File: src/components/LeftNav/TopLinkItem/TopLinkItem.tsx [hunk 1]
```diff
@@ -50,7 +53,8 @@ export const TopLinkItem = ({
     <li
       className={clsx(
         "neo-leftnav__main",
-        isActive && "neo-leftnav__main--active"
+        isActive && "neo-leftnav__main--active",
+        className
```
Comment (enrique-prado) [inline src/components/LeftNav/TopLinkItem/TopLinkItem.tsx:57]: Looks good but you'll need to add similar logic to the LinkItem component (which are inside categories) But you can do that as a separate ticket.

**Review by enrique-prado — APPROVED** [review]
Left one comment.

**Comment by mrazian85** [comment]
> you deleted all the badges? that's a no-no

Oh whoops. I did not catch that. Let me see what went wrong there

**Comment by mrazian85** [comment]
@joe-s-avaya badges are back! And 0.1% higher coverage ✨

File: src/components/LeftNav/TopLinkItem/TopLinkItem.tsx (hunk 1 above)
Comment (mrazian85) [reply src/components/LeftNav/TopLinkItem/TopLinkItem.tsx:57]: Thanks Enrique!

Thought it looked to me like the `LinkItem` can already accept an `active` prop. I assumed we should already be able to override the active state that way? It also accepts a `className` prop

Comment (enrique-prado) [reply src/components/LeftNav/TopLinkItem/TopLinkItem.tsx:57]: Ok, that's perfect. You can ignore my comment then.

**Review by joe-s-avaya — APPROVED** [review]
this feels real hacky to me, but I don't have an improvement to suggest, so I'll approve, but it feels like something we may want to re-work in the future

**Pull request merged**